auth/src/main/java/com/firebase/ui/auth/ui/phone/PhoneActivity.java

```diff
@@ -14,401 +14,204 @@
-
-        getAuthHelper().getPhoneAuthProvider().verifyPhoneNumber(
-                phoneNumber,
-                AUTO_RETRIEVAL_TIMEOUT_MILLIS,
-                TimeUnit.MILLISECONDS,
-                this,
-                new PhoneAuthProvider.OnVerificationStateChangedCallbacks() {
-                    @Override
-                    public void onVerificationCompleted(@NonNull PhoneAuthCredential phoneAuthCredential) {
-                        if (!mIsDestroyed) {
-                            PhoneActivity.this.onVerificationSuccess(phoneAuthCredential);
-                        }
-                    }
-
-                    @Override
-                    public void onVerificationFailed(FirebaseException ex) {
-                        if (!mIsDestroyed) {
-                            PhoneActivity.this.onVerificationFailed(ex);
-                        }
-                    }
-
-                    @Override
-                    public void onCodeSent(@NonNull String verificationId,
-                                           @NonNull PhoneAuthProvider.ForceResendingToken forceResendingToken) {
-                        mVerificationId = verificationId;
-                        mForceResendingToken = forceResendingToken;
-                        if (!mIsDestroyed) {
-                            PhoneActivity.this.onCodeSent();
-                        }
-                    }
-                },
-                forceResend ? mForceResendingToken : null);
-    }
-
-    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
-    protected AlertDialog getAlertDialog() {
-        // It is hard to test AlertDialogs currently with robo electric. See:
-        // https://github.com/robolectric/robolectric/issues/1944
-        // We just test that the error was not displayed inline
-        return mAlertDialog;
+    private void showSubmitCodeFragment(String number) {
+        getSupportFragmentManager().beginTransaction()
+                .replace(
+                        R.id.fragment_phone,
+                        SubmitConfirmationCodeFragment.newInstance(number),
+                        SubmitConfirmationCodeFragment.TAG)
+                .addToBackStack(null)
+                .commit();
     }
 
-    private void showSubmitCodeFragment() {
-        // idempotent function
-        if (getSubmitConfirmationCodeFragment() == null) {
-            SubmitConfirmationCodeFragment f = SubmitConfirmationCodeFragment.newInstance(
-                    getFlowParams(), mPhoneNumber);
-            FragmentTransaction t = getSupportFragmentManager().beginTransaction()
-                    .replace(R.id.fragment_verify_phone, f, SubmitConfirmationCodeFragment.TAG)
-                    .addToBackStack(null);
-
-            if (!isFinishing() && !mIsDestroyed) {
-                t.commitAllowingStateLoss();
-            }
+    @Override
+    public void showProgress(int message) {
+        FragmentBase fragment = (CheckPhoneNumberFragment)
```

Comment:
Could we DRY up the first few lines from `showProgress` and `hideProgress` into some `getActiveFragment` method or something?